haskell-src/Concordium/Constants.hs

```diff
@@ -19,9 +19,7 @@ maxPayloadSize :: SProtocolVersion pv -> Word32
 maxPayloadSize SP1 = 100 * 1024 -- 100kB
 maxPayloadSize SP2 = 100 * 1024 -- 100kB
 maxPayloadSize SP3 = 100 * 1024 -- 100kB
-maxPayloadSize SP4 = maxWasmModuleSizeV1 + 1 + 4 + 4 -- +1 for the payload tag, +4 for the length, +4 for the module version
-maxPayloadSize SP5 = maxPayloadSize SP4
-maxPayloadSize SP6 = maxPayloadSize SP4
+maxPayloadSize _ = maxWasmModuleSizeV1 + 1 + 4 + 4 -- +1 for the payload tag, +4 for the length, +4 for the module version
```

Comment:
I don't really like this. I prefer that it is explicit and warned when new protocols are defined. So we don't miss something like this when we need to change it.